### ISSUE TITLE
Message objects should have some convenience debug print functionality

### ISSUE THREAD

**Issue by richteer** [Halibot/halibot#97]
When testing, it is pretty useful to get a dump of a `Message` object's information, so there should probably be some default implementation of this. Pasted below is an implementation of `__str__()` that I've been using myself when playing around. Use this thread for discussion on what might be the best method.

```diff
diff --git a/halibot/message.py b/halibot/message.py
index a87befd..5014658 100644
--- a/halibot/message.py
+++ b/halibot/message.py
@@ -19,3 +19,6 @@ class Message():
 
        def whom(self):
                return '/'.join(self.target.split('/')[1:])
+
+       def __str__(self):
+               return "body='{}' type='{}' author='{}' origin='{}' target='{}'".format(self.body, self.type, self.author, self.origin, self.target)
```

**Comment by richteer** [comment]
`__str__()` allows for a message to be directly `print()`'d or logged, which is quite convenient. However, I'm not sure what precedent we should be setting here. Maybe there should be a function that must be manually called to explicitly request this string? That way there is no ambiguity between accessing this string and something like `msg.body`?

Furthermore, this implementation could be a lot simplier if `Message()` was inherited `jsdict`, however that blows some stuff up pretty bad.

Finally, some keys are ignored, like `identity`. Might make sense to drop fields that are `None`, so we don't clutter all prints with an empty `identity` or something.

Second question: should we use this or `__repr__()`, what the heck is the difference again...

**Comment by richteer** [comment]
Removing the RFC tag, I think the above proposal is "good enough" to at least get something in core.

**Comment by mracine** [comment]
Taking a look at this, perhaps both a __repr__ and __str__ function could be useful depending on the context.

**Comment by richteer** [comment]
I agree, `__repr__()` seems to be the more proper way to dump it in a log, but maybe we could make `__str__()` redirect to `msg.body` or something? Is that a safe assumption, or should we discuss that further?

Regardless, I think the `__repr__()` change won't have any contention, feel free to PR that one for sure.

**Comment by richteer** [comment]
Merged with #123
